src/libs/ReportUtils.ts

```diff
@@ -1364,49 +1442,42 @@ function getReimbursementQueuedActionMessage(reportAction, report) {
  *
- * @param {String} reportID - the report for which last visible message has to be fetched
- * @param {Object} [actionsToMerge] - the optimistic merge actions that needs to be considered while fetching last visible message
- * @return {Object}
+ * @param reportID - the report for which last visible message has to be fetched
+ * @param [actionsToMerge] - the optimistic merge actions that needs to be considered while fetching last visible message
+
  */
-function getLastVisibleMessage(reportID, actionsToMerge = {}) {
+function getLastVisibleMessage(reportID: string | undefined, actionsToMerge: ReportActions = {}): LastVisibleMessage {
     const report = getReport(reportID);
-    const lastVisibleAction = ReportActionsUtils.getLastVisibleAction(reportID, actionsToMerge);
+    const lastVisibleAction = ReportActionsUtils.getLastVisibleAction(reportID ?? '', actionsToMerge);
 
     // For Chat Report with deleted parent actions, let us fetch the correct message
-    if (ReportActionsUtils.isDeletedParentAction(lastVisibleAction) && isChatReport(report)) {
+    if (ReportActionsUtils.isDeletedParentAction(lastVisibleAction) && report && isNotEmptyObject(report) && isChatReport(report)) {
         const lastMessageText = getDeletedParentActionMessageForChatReport(lastVisibleAction);
         return {
             lastMessageText,
         };
     }
 
     // Fetch the last visible message for report represented by reportID and based on actions to merge.
-    return ReportActionsUtils.getLastVisibleMessage(reportID, actionsToMerge);
+    return ReportActionsUtils.getLastVisibleMessage(reportID ?? '', actionsToMerge);
 }
 
 /**
  * Checks if a report is an open task report assigned to current user.
  *
- * @param {Object} report
- * @param {Object} [parentReportAction] - The parent report action of the report (Used to check if the task has been canceled)
- * @returns {Boolean}
+ * @param [parentReportAction] - The parent report action of the report (Used to check if the task has been canceled)
  */
-function isWaitingForAssigneeToCompleteTask(report, parentReportAction = {}) {
+function isWaitingForAssigneeToCompleteTask(report: OnyxEntry<Report>, parentReportAction: OnyxEntry<ReportAction> | EmptyObject = {}): boolean {
     return isTaskReport(report) && isReportManager(report) && isOpenTaskReport(report, parentReportAction);
 }
 
-/**
- * @param {Object} report
- * @returns {Boolean}
- */
-function isUnreadWithMention(report) {
+function isUnreadWithMention(report: OnyxEntry<Report> | OptionData): boolean {
     if (!report) {
         return false;
     }
-
     // lastMentionedTime and lastReadTime are both datetime strings and can be compared directly
-    const lastMentionedTime = report.lastMentionedTime || '';
-    const lastReadTime = report.lastReadTime || '';
+    const lastMentionedTime = report?.lastMentionedTime ?? '';
+    const lastReadTime = report?.lastReadTime ?? '';
```

Comment:
```suggestion
    const lastMentionedTime = report.lastMentionedTime ?? '';
    const lastReadTime = report.lastReadTime ?? '';
```
Optional chain is not required here as report is not null now